app/controllers/company.controller.js

```diff
@@ -0,0 +1,130 @@
+const db = require('../models');
+
+const {
+  Company,
+  User,
+  Sequelize: { Op },
+} = db;
+
+// Create a company
+exports.createCompany = async (req, res, next) => {
+  const {
+    body: {
+      userId,
+      companyName,
+      website,
+      primaryColor,
+      secondaryColor,
+      address,
+      logo,
+      bio,
+    },
+  } = req;
+
+  User.findById(userId)
+    .then((user) => {
+      user.update({ isCompanyOwner: true });
+      return Company.findOrCreate({
+        where: {
+          [Op.or]: [{ companyName }, { website }],
+        },
+        defaults: {
+          companyName,
+          website,
+          ownerUserId: userId,
+          primaryColor: primaryColor || '#000000',
+          secondaryColor: secondaryColor || '#000000',
+          address: address || '',
+          logo: logo || 'https://pbs.twimg.com/media/FB6YhR8WQAI5MnM.png',
+          bio: bio || '',
+          createdAt: Date.now(),
+        },
+      });
+    })
+    .then(([company, created]) => {
+      if (!created) {
+        return res.status(400).send({
+          message: 'Company already exists',
+        });
+      }
+      return res.status(200).send({
+        message: 'Company created successfully',
+        data: company,
+      });
+    })
+    .catch((err) => next(err));
+};
+
+// Delete a company
+exports.deleteCompany = async (req, res, next) => {
+  const {
+    body: { id, userId },
+  } = req;
+
+  Company.findById(id)
+    .then((company) => {
+      if (company.ownerUserId !== userId) {
+        return next(new StatusError('unauthorized', 401));
+      }
+      return company.destroy();
+    })
+    .then(async () => {
+      const owner = await Company.findOne({
+        where: {
+          ownerUserId: userId,
+        },
+      });
+      if (owner === null) {
+        User.findById(userId).then((user) => {
+          user.update({ isCompanyOwner: false });
+        });
+      }
+    })
```

Comment:
so first you destroy the company with `id` in the body, making sure that the owner is `userId` which is good. then you check if there are other companies which the user owns and then if not you set their `isCompanyOwner` field to false. 

You seem to have defined a one to one relationship between `company` and `user`. Why do you expect a user to have many companies?

Reply:
Yeah, i thought for the future it might be better to be able to create multiple companies. But you are right, lets keep the one to one relationship for now.